internal/mode/static/state/graph/route_common.go

```diff
@@ -63,6 +69,27 @@ type RouteKey struct {
 	RouteType RouteType
 }
 
+type L4Route struct {
```

Comment:
As you work on the next piece of TLSRoute, see if there's a way to extract the common elements of the L4 and L7 routes into a common type. I'm noticing there's a lot of overlap, for example, building the upstreams is almost identical. It would be great if we could reduce the code duplication.